packages/vite-node/src/cli.ts

```diff
@@ -81,3 +89,23 @@ async function run(files: string[], options: CliOptions = {}) {
       await runner.executeFile(file)
   })
 }
+
+function parseServerOptions(serverOptions: ViteNodeServerOptions) {
+  if (serverOptions.deps && serverOptions.deps.inline) {
+    serverOptions.deps.inline = serverOptions.deps.inline.map((dep) => {
+      return typeof dep === 'string' && dep.startsWith('/') && dep.endsWith('/')
+        ? new RegExp(dep)
+        : dep
+    })
+  }
+
+  if (serverOptions.deps && serverOptions.deps.external) {
+    serverOptions.deps.external = serverOptions.deps.external.map((dep) => {
+      return typeof dep === 'string' && dep.startsWith('/') && dep.endsWith('/')
+        ? new RegExp(dep)
+        : dep
+    })
+  }
+
+  // TODO: Handle serverOptions.transformMode.
```

Comment:
I've noticed that `serverOptions.transformMode.ssr` (and `.web`) only accept Regexes, while `serverOptions.deps.inline` (and `.external`) accepts both strings and Regexes.

- Is that expected?
- If so, would you mind giving a bit of context as why? 🙏

Reply:
doesn't have a reason 👀 
I guess just didn't notice. but to be honest I prefer using regexp always, because current solution for `deps.inline` as a string is not reliable on non-npm/pnpm package managers